### PR TITLE
Periodic value deserialization

### LINKED ISSUE #115: Add multi-period parameters to deserialization
In the multi-period approach, it should be possible to pass period-specific parameters. It's possible to just pass a sequence, but to reduce data we would like to solve this internally.

How this is done depends on the feedback on `oemof.solph` [Issue#959](https://github.com/oemof/oemof-solph/issues/959) and first draft is done in [PR#960](https://github.com/oemof/oemof-solph/pull/960)

- [x] <s> refactor deserialization </s>
- [x] read period info first
- [x] read lists and compare to number of periods/length
- [x] convert to full list <s>or plumbing </s>
- [x] <s> catch non-periodic parameters </s>

- This does not work for all parameters, especially the ones which go into a flow, e.g. capacity/nominal_value
- This does also not work for capacity_costs/ep_costs as passed lists are iterated over per period [here](https://github.com/oemof/oemof-solph/blob/3350f5adc349ef12a58a0d5e0e9d93fe7306c328/src/oemof/solph/flows/_investment_flow_block.py#L855)

### PR DESCRIPTION
**Pull request by nailend**
close #115 

if multi-period approach is taken (periods resource exists):
- facade parameters are checked if list and length equals number of periods
- a full time series is generated with period specific values

This is only possible for certain parameters, otherwise building the energy system fails.
- warning is raised every time periodic values are converted to a timeseries

**Attention:**

The investment parameters work differently. One period specific value is enough, and a whole timeseries is not necessary. Check [here](https://github.com/oemof/oemof-solph/blob/3350f5adc349ef12a58a0d5e0e9d93fe7306c328/src/oemof/solph/flows/_investment_flow_block.py#L855)

e.g. "**fixed_costs**" "**capacity_costs**"

These period values are handled differently:
- no timeseries is created
- only list of values for every period are passed

### CODE REVIEW

File: src/oemof/tabular/datapackage/reading.py [hunk 1]
```diff
@@ -377,6 +377,56 @@ def find(n, d):
...
+        periods_index : list
+            List containing periods datetimeindex.
+
+        Returns
+        -------
+        list
+            List of periodic values.
+        """
+        # check if length of list equals number of periods
+        if len(values) == len(periods_index):
```
Comment (henhuy) [inline src/oemof/tabular/datapackage/reading.py]: would check for inequality and remove pass statement

**Comment by iuljan** [comment]
Change in [oemof.solph issue 983](https://github.com/oemof/oemof-solph/issues/983) might have an effect on this

File: src/oemof/tabular/datapackage/reading.py [hunk 2]
```diff
@@ -405,6 +455,37 @@ def find(n, d):
...
+                                facade[f] = v
+                                msg = (
+                                    f"\nThe parameter '{f}' of a "
+                                    f"'{facade['type']}' facade is converted "
+                                    "into a periodic list. This might not be "
+                                    "possible for every parameter and lead to "
+                                    "ambiguous error messages.\nPlease be "
+                                    "aware, when using this feature!"
+                                )
+                                warnings.warn(msg, UserWarning)
```
Comment (henhuy) [inline src/oemof/tabular/datapackage/reading.py]: The message does not belong here, as paramter has not been converted.
Instead, I would replace whole part with simple `continue` as value does not has to be adapted...

**Review by henhuy — CHANGES_REQUESTED** [review]
Minor change requested. Otherwise this looks good to me.

**Comment by iuljan** [comment]
@henhuy not sure what's happening here and why the tests fail. Locally, there is no error ... Could you please check locally as well?

**Comment by henhuy** [comment]
I could reproduce error with latest python version 3.9.18 - in 3.9.16 error did not occur.
Could not figure out what has been changed in between those versions, though.
Error comes up, as `Flow` component "has" an attribute `_in_edges`, but for some reasons the attribute cannot be accessed and therefor throws an error:
https://github.com/oemof/oemof-solph/blob/f02297ea7db0134bfff12c3ff6c7c909ca4405ed/src/oemof/solph/processing.py#L515-L532
Fix could be to reverse checking:
```
attrs = [
    i
    for i in dir(com)
    if not (i.startswith(exclusions) or callable(getattr(com, i)))
]
```
We could implement this fix in `oemof.solph` - but first we would have to add a failing test to show and explain what's happening.

**Comment by iuljan** [comment]
Strange... python version 3.9.18 is from August'23 and was also already used in this [CI run](https://github.com/oemof/oemof-tabular/actions/runs/6422187115/job/17438171580) which wasn't failing.

**Comment by iuljan** [comment]
I found it. It's caused by changes in  `oemof.network` apparently version `0.5.0a1` was still working.
This might be of interest to you, @p-snft ?

**Comment by p-snft** [comment]
Unfortunately, network has some untested and undocumented parts. I did a release that did not really change anything to the "official" API.

I already reversed the checks that @henhuy mentioned in dev branch of solph: https://github.com/oemof/oemof-solph/commit/559c6811d33c5f34d0363071539742b37686153e. Probably, a release makes sense as well.

**Comment by p-snft** [comment]
There now is [oemof.solph 0.5.2.dev0](https://pypi.org/project/oemof.solph/0.5.2.dev0/) that should fix the issue.

**Pull request merged**